Code/GraphMol/RascalMCES/RascalResult.cpp

```diff
@@ -0,0 +1,817 @@
+    }
+  }
+  for (auto atom : tmpMol.atoms()) {
+    if (mol1Atoms.find(atom->getIdx()) == mol1Atoms.end()) {
+      tmpMol.removeAtom(atom);
+    }
+  }
+  tmpMol.commitBatchEdit();
+  MolOps::removeHs(tmpMol);
+  MolOps::sanitizeMol(tmpMol);
+  d_mcesMol.reset(new ROMol(tmpMol));
+  return d_mcesMol;
+}
+
+bool resultCompare(const RascalResult &res1, const RascalResult &res2) {
+  if (res1.getBondMatches().size() == res2.getBondMatches().size()) {
+    if (res1.getNumFrags() == res2.getNumFrags()) {
+      if (res1.getLargestFragSize() == res2.getLargestFragSize()) {
+        if (res1.getRingNonRingBondScore() == res2.getRingNonRingBondScore()) {
+          if (res1.getAtomMatchScore() == res2.getAtomMatchScore()) {
+            if (res1.getMaxDeltaAtomAtomDist() ==
+                res2.getMaxDeltaAtomAtomDist()) {
+              return res1.getSmarts() < res2.getSmarts();
+            } else {
+              return res1.getMaxDeltaAtomAtomDist() <
+                     res2.getMaxDeltaAtomAtomDist();
+            }
+          } else {
+            return res1.getAtomMatchScore() < res2.getAtomMatchScore();
+          }
+        } else {
+          return res1.getRingNonRingBondScore() <
+                 res2.getRingNonRingBondScore();
+        }
+      } else {
+        return res1.getLargestFragSize() > res2.getLargestFragSize();
+      }
+    } else {
+      return res1.getNumFrags() < res2.getNumFrags();
+    }
+  } else {
+    return res1.getBondMatches().size() > res2.getBondMatches().size();
+  }
+}
+
+void extractClique(const std::vector<unsigned int> &clique,
+                   const std::vector<std::pair<int, int>> &vtxPairs,
+                   bool swapped,
+                   std::vector<std::pair<int, int>> &bondMatches) {
+  bondMatches.clear();
+  for (auto mem : clique) {
+    if (swapped) {
+      bondMatches.push_back(
+          std::make_pair(vtxPairs[mem].second, vtxPairs[mem].first));
+    } else {
+      bondMatches.push_back(
+          std::make_pair(vtxPairs[mem].first, vtxPairs[mem].second));
+    }
+  }
+  std::sort(bondMatches.begin(), bondMatches.end());
+}
+
+void cleanSmarts(std::string &smarts) {
+  const static std::vector<std::pair<std::regex, std::string>> repls{
+      {std::regex(R"(\[#6&A\])"), "C"},
+      {std::regex(R"(\[#6&A&R\])"), "[C&R]"},
+      {std::regex(R"(\[#6&a\])"), "c"},
+      {std::regex(R"(\[#7&A\])"), "N"},
+      {std::regex(R"(\[#7&A&R\])"), "[N&R]"},
+      {std::regex(R"(\[#7&a\])"), "n"},
+      {std::regex(R"(\[#8&A\])"), "O"},
+      {std::regex(R"(\[#8&A&R\])"), "[O&R]"},
+      {std::regex(R"(\[#8&a\])"), "o"},
+      {std::regex(R"(\[#9&A\])"), "F"},
+      {std::regex(R"(\[#16&A\])"), "S"},
+      {std::regex(R"(\[#16&a\])"), "s"},
+      {std::regex(R"(\[#17&A\])"), "Cl"},
+      {std::regex(R"(\[#35&A\])"), "Br"},
+      {std::regex(R"(\[#53&A\])"), "I"},
+      {std::regex(R"(([cnops][1-9]*):([cnops]))"), "$1$2"},
```

Comment:
Good point for the first 2, with aromatic atoms. It does leave you with ugly things like c1:c:c:c:c:c:1 which I was trying to avoid.  Maybe the SMARTS writer is the place to address that.
The remaining ones are a single bond to an aliphatic atom.  The default for that is always single, isn't it?

Reply:
yeah, the bonds to aliphatic atoms need to be single.